### PR TITLE
Vault secrets gateway pools

### PR DESCRIPTION
**Pull request by hghaf099**
### Changes proposed in this PR:
Adding a new subcommand for vault-secrets. This subcommands helps with managing gateway pools. It has the usual CRUD operations. 

### How I've tested this PR:
I have built the binary and run the new commands against the dev environment. 

image

### Checklist:
- [ ] Tests added if applicable
- [x] CHANGELOG entry added or label 'pr/no-changelog' added to PR
  > Run `CHANGELOG_PR=<PR number> make changelog/new-entry` for guidance
  > in authoring a changelog entry, and commit the resulting file, which should
  > have a name matching your PR number. Entries should use imperative present
  > tense (e.g. Add support for...)

### CODE REVIEW

**Review by ncabatoff — APPROVED** [review]
We'd planned in the RFC to have the `read` command also include the list of integrations obtained via ListGatewayPoolIntegrations.  I don't mind whether it's included in this PR or not, your call.

File: internal/commands/vaultsecrets/gatewaypools/update.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			},
+		},
+		Flags: cmd.Flags{
+			Local: []*cmd.Flag{
+				{
+					Name:         "description",
+					DisplayValue: "DESCRIPTION",
+					Description:  "The updated gateway pool description.",
+					Value:        flagvalue.Simple("", &opts.Description),
+					Required:     true,
```
[the remark this answers is not in the document]
Comment (hghaf099) [reply internal/commands/vaultsecrets/gatewaypools/update.go:69]: Hmm, so we have this required in the validation layer. Should I be removing that from there as well? 

https://github.com/hashicorp/cloud-vault-secrets/blob/9d45211cff964087f6651da9ab9ce40c91619c7d/internal/service/validation/service.go#L1965

[the remark this answers is not in the document]
Comment (ncabatoff) [reply internal/commands/vaultsecrets/gatewaypools/update.go:69]: We debated that as well IIRC, let's not revisit that just yet.  I was thinking more that the CLI could return an error saying "no fields specified for update" if (optional) description isn't specified, and then if/when we add another updatable field, that logic wouldn't have to change other than to add another condition to the `if` for the new field.

Totally optional suggestion btw, it's also ok the way you have it.

[the remark this answers is not in the document]
Comment (hghaf099) [reply internal/commands/vaultsecrets/gatewaypools/update.go]: Ah, I see. Fixed in https://github.com/hashicorp/hcp/pull/131/commits/bb549ffc3e687cbe531e5cd1bb5367a1d43eeb3f

**Pull request merged**